src/components/app/ProjectPreview.svelte

```diff
@@ -31,6 +37,42 @@
         showCollaborators?: boolean;
     }
 
+    function findCharacterName(value: Value): string | null {
+        // If it's a MarkupValue, check for character links
+        if (value instanceof MarkupValue) {
+            const nodes = value.markup.nodes();
+            for (const node of nodes) {
+                if (node instanceof ConceptLink) {
+                    const parsed = ConceptLink.parse(node.getName());
+                    if (parsed instanceof CharacterName) {
+                        return `${parsed.username}/${parsed.name}`;
+                    }
+                }
+            }
+        }
+
+        // If it's a StructureValue, check all its fields recursively
+        if (
+            value &&
+            value.constructor &&
+            value.constructor.name === 'StructureValue' &&
+            (value as any).context &&
```

Comment:
Never use `any` in TypeScript. This expression should be checking if the value is an instance of `StructureValue` (e.g., "value instanceof StructureValue"). That is the type safe way to verify a value's type.